src/yunohost/log.py

```diff
@@ -0,0 +1,424 @@
+
+            result[category].append(entry)
+
+    return result
+
+
+def log_display(path, number=50):
+    """
+    Display a log file enriched with metadata if any.
+
+    If the file_name is not an absolute path, it will try to search the file in
+    the unit operations log path (see OPERATIONS_PATH).
+
+    Argument:
+        file_name
+        number
+    """
+
+    # Normalize log/metadata paths and filenames
+    abs_path = path
+    log_path = None
+    if not path.startswith('/'):
+        for category in CATEGORIES:
+            abs_path = os.path.join(CATEGORIES_PATH, category, path)
+            if os.path.exists(abs_path) or os.path.exists(abs_path + METADATA_FILE_EXT):
+                break
+
+    if os.path.exists(abs_path) and not path.endswith(METADATA_FILE_EXT) :
+        log_path = abs_path
+
+    if abs_path.endswith(METADATA_FILE_EXT) or abs_path.endswith(LOG_FILE_EXT):
+        base_path = ''.join(os.path.splitext(abs_path)[:-1])
+    else:
+        base_path = abs_path
+    base_filename = os.path.basename(base_path)
+    md_path = base_path + METADATA_FILE_EXT
+    if log_path is None:
+        log_path = base_path + LOG_FILE_EXT
+
+    if not os.path.exists(md_path) and not os.path.exists(log_path):
+        raise MoulinetteError(errno.EINVAL,
+                              m18n.n('log_does_exists', log=path))
+
+    infos = {}
+
+    # If it's a unit operation, display the name and the description
+    if base_path.startswith(CATEGORIES_PATH):
+        infos["description"] = _get_description_from_name(base_filename)
+        infos['name'] = base_filename
+
+    # Display metadata if exist
+    if os.path.exists(md_path):
+        with open(md_path, "r") as md_file:
+            try:
+                metadata = yaml.safe_load(md_file)
+                infos['metadata_path'] = md_path
+                infos['metadata'] = metadata
+                if 'log_path' in metadata:
+                    log_path = metadata['log_path']
+            except yaml.YAMLError:
+                error = m18n.n('log_corrupted_md_file', file=md_path)
+                if os.path.exists(log_path):
+                    logger.warning(error)
+                else:
+                    raise MoulinetteError(errno.EINVAL, error)
+
+    # Display logs if exist
+    if os.path.exists(log_path):
+        from yunohost.service import _tail
+        logs = _tail(log_path, int(number))
+        #logs = [{"datetime": x.split(": ", 1)[0].replace("_", " "), "line": x.split(": ", 1)[1]}  for x in logs if x]
+        infos['log_path'] = log_path
+        infos['logs'] = logs
+
+    return infos
+
+def is_unit_operation(entities='app,domain,service,user', exclude='auth,password', operation_key=None, auto=True):
+    """
+    Configure quickly a unit operation
+
+    This decorator help you to configure quickly the record of a unit operations.
+
+    Argument:
+    entities    A list seperated by coma of entity types related to the unit
```

Comment:
Why a list of comma separated entities in a string and not a python list? (or a tuple) It's meant for that :yum:

Reply:
+1